### PR TITLE
Plugins: Disable update button when cloud install is not completed

### PR DESCRIPTION
**Pull request by oshirohugo**
**What is this feature?**

Disable the update plugin button when cloud installation is not completed, which means, the grafana-com instance has the plugin but the instance with the new plugin isn't running yet.

**Why do we need this feature?**

This is needed to so the user doesn't see the update button enabled again when the update isn't completed. This will prevent the user from calling update multiple times and errors to occur.

**Who is this feature for?**

Plugin catalog users

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/plugins/admin/helpers.ts [hunk 1]
```diff
@@ -49,8 +49,15 @@ export function mergeLocalsAndRemotes({
...
-        catalogPlugin.isInstalled = instancesSet.has(remotePlugin.slug) || catalogPlugin.isInstalled;
+          : instancesMap.has(remotePlugin.slug) && catalogPlugin.isInstalled;
+
+        catalogPlugin.isInstalled = instancesMap.has(remotePlugin.slug) || catalogPlugin.isInstalled;
+
+        const instancePlugin = instancesMap.get(remotePlugin.slug);
+        catalogPlugin.isUpdating =
+          instancesMap.has(remotePlugin.slug) &&
+          catalogPlugin.hasUpdate &&
+          catalogPlugin.installedVersion !== instancePlugin?.version;
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/helpers.ts:60]: I think we should make the name `isUpdating` a bit more specific / clear, as if I understand correctly it is only used for checking if the plugin is being updated _in cloud_ (in an on-prem instance this is going to be `false` even if the plugin is being updated), am I right?

**Review by leventebalogh — COMMENTED** [review]
Nice work getting these scenarios sorted! 👏

File: public/app/features/plugins/admin/helpers.ts (hunk 1 above)
Comment (oshirohugo) [reply public/app/features/plugins/admin/helpers.ts:60]: Exactly, I think the same comment that you did [here](https://github.com/grafana/grafana/pull/81907#discussion_r1482553436) applies for this case.

I will change the variable name accordingly. :smile:

**Comment by oshirohugo** [comment]
@leventebalogh coudl you have a second look on this one?

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx [hunk 2]
```diff
@@ -90,4 +92,79 @@ describe('InstallControlsButton', () => {
     );
     expect(screen.queryByRole('button')).not.toBeInTheDocument();
   });
+
+  describe('update button on prem', () => {
+    const store = configureStore({
+      plugins: getPluginsStateMock([]),
+    });
+
+    it('should be disable when is Installing', () => {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx:101]: Typo
```suggestion
    it('should be disabled when is Installing', () => {
```

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx [hunk 3]
```diff
@@ -90,4 +92,79 @@ describe('InstallControlsButton', () => {
...
+      render(
+        <TestProvider store={store}>
+          <InstallControlsButton plugin={{ ...plugin }} pluginStatus={PluginStatus.UPDATE} />
+        </TestProvider>
+      );
+      const button = screen.getByText('Updating').closest('button');
+      expect(button).toBeDisabled();
+    });
+
+    it('should be enabled when not is Installing', () => {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx:112]:
```suggestion
it('should be enabled when it is not installing', () => {
```

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx [hunk 4]
```diff
@@ -90,4 +92,79 @@ describe('InstallControlsButton', () => {
...
+        <TestProvider store={store}>
+          <InstallControlsButton plugin={{ ...plugin }} pluginStatus={PluginStatus.UPDATE} />
+        </TestProvider>
+      );
+      const button = screen.getByText('Update').closest('button');
+      expect(button).toBeEnabled();
+    });
+  });
+
+  describe('update button on manage instance', () => {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx:124]:
```suggestion
describe('update button on managed instance', () => {
```

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx [hunk 5]
```diff
@@ -90,4 +92,79 @@ describe('InstallControlsButton', () => {
...
+    afterAll(() => {
+      config.featureToggles.managedPluginsInstall = oldFeatureTogglesManagedPluginsInstall;
+      config.pluginAdminExternalManageEnabled = oldPluginAdminExternalManageEnabled;
+    });
+
+    const store = configureStore({
+      plugins: getPluginsStateMock([]),
+    });
+
+    it('should be disabled when is Installing=false but isUpdatingFromInstance=true', () => {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx:142]:
```suggestion
it('should be disabled when isInstalling=false but isUpdatingFromInstance=true', () => {
```

File: public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx [hunk 6]
```diff
@@ -90,4 +92,79 @@ describe('InstallControlsButton', () => {
...
+            plugin={{ ...plugin, isUpdatingFromInstance: true }}
+            pluginStatus={PluginStatus.UPDATE}
+          />
+        </TestProvider>
+      );
+      const button = screen.getByText('Update').closest('button');
+      expect(button).toBeDisabled();
+    });
+
+    it('should be disabled when is Installing=false but isUpdatingFromInstance=false', () => {
```
Comment (leventebalogh) [inline public/app/features/plugins/admin/components/InstallControls/InstallControlsButton.test.tsx:156]:
```suggestion
it('should be enabled when is Installing=false and isUpdatingFromInstance=false', () => {
```

**Review by leventebalogh — APPROVED** [review]

**Pull request merged**